### ISSUE TITLE
"'path\<deserialized_file>' already exists" error while trying to execute

### ISSUE THREAD

**Issue by EricDraven2503** [AppleSheeple/telegram-media-deserialize#2]
Hi, I don't seem to understand how the usage of this tool works.
According to the infos, after cache files have been decrypted, it's supposed to take as first parameter the serialized file of the two in the same folder of the decrypted files, while taking as second parameter the other one.
I tried to do it and even swapping them, but in both cases it returns me the error as from title.
What am I doing wrong?

**Comment by AppleSheeple** [comment]
> Hi, I don't seem to understand how the usage of this tool works.

Yeah, apologies. The instructions are indeed probably not average-user friendly.

> it's supposed to take as first parameter the serialized file of the two in the same folder of the decrypted files, while taking as second parameter the other one.

```
telegram-media-deserialize <serialized_file> <deserialized_file>
```

`<deserialized_file>` represents the name of a new file to be created which contains the result of deserialisation.

That file should be truncated (e.g. using `truncate -s` in GNU/Linux) or partially copied (e.g. using `dd`) with the *last contiguous offset* value in mind. Then the following non-serialized files can be appended to that.

So, suppose a media file is split between the decrypted files:
 * `file1` (serialized)
 * `file2` (non-serialized)
 * `file3` (non-serialized)

Then:

```
telegram-media-deserialize file1 file
```

Suppose *last contiguous offset* is 8388608, then

```
truncate -s 8388608 file
cat file2 >> file
cat file3 >> file
```

And `file` should now be the full desired media file.

**Comment by EricDraven2503** [comment]
After my decryption, I've been crossing even media files split between two or three serialized files, but no non-serialized ones.
They all have an exact size of 8.193 KB, and they all seem to be unreadable.
Is there any way to read them or are they corrupted ones?

**Comment by EricDraven2503** [comment]
After my decryption, I've been encountering even media files split between two or three serialized files, but no non-serialized ones.
They all have an exact size of 8.193 KB, and they all seem to be unreadable.
Is there any way to read them or are they corrupted ones?

**Comment by AppleSheeple** [comment]
@EricDraven2503 

If videos are not played completely in the app, then partial content can indeed be stored in individual serialized files. But those should be playable after deserialisation, even if they are partial.

The same video won't be, as far as I know, contained in multiple serialized files as I already mentioned.

Also, beware of how to get the order of files correctly. See #1 for a discussion about that.

If I had to guess, the files you mention are probably not serialized. They wouldn't be playable as they need to be appended in the right order to the right deserialised and truncated file (or files if they belong to different videos). They would be `file2`, `file3`,.... in the example in my previous comment.
